design/object-bucket/object-bucket.md

```diff
@@ -3,57 +3,47 @@
+Currently, the Rook-Ceph operator enables the dynamic creation of Ceph cluster, object stores, and users via Rook native Custom Resource Definitions.  This proposal introduces the next logical step in this design: the addition a controller for handling dynamic bucket creation by leveraging the [lib-bucket-provisioner library](https://github.com/yard-turkey/lib-bucket-provisioner).  This library defines two new CRDs: ObjectBucketClaim and ObjectBucket.  Additionally, the library wraps a Kubernetes to watch related API events.
 
 ## Glossary
 
-- Admin: A Kubernetes cluster administrator with RBAC permissions to instantiate and control access to cluster storage. The admin will provision object stores via CephObjectStore custom resources and create StorageClasses to enable user access.  
-- User: A Kubernetes cluster user with limited permissions, typically confined to CRUD operations on Kubernetes objects within a single namespace.  The user will request Ceph Object buckets by instantiating ObjectBucketClaims.
-- ObjectBucketClaim: a namespaced, user facing Custom Resource Definition representing a request for a dynamically created bucket.  ObjectBucketClaims reference a StorageClass whose _provisioner_ field contains the name of an object store provisioner capable of dynamically provisioning Ceph-RGW buckets.
-- ObjectBucket: a non-namespace, admin facing Custom Resource Definition representing the fulfillment of a user request for a dynamically created bucket in an existing object store (similar in use to PersistentVolumes).  ObjectBuckets serve to provide discrete information about the bucket and object store provider to admins.
+- Admin: A Kubernetes cluster administrator with RBAC permissions to instantiate and control access to cluster storage. The admin  provisions object stores via CephObjectStore custom resources and creates StorageClasses to enable user access.  
+- User: A Kubernetes cluster user with limited permissions, typically confined to CRUD operations on Kubernetes objects within a single namespace.  The user will request Ceph buckets by instantiating ObjectBucketClaims.
+- ObjectBucketClaim: a namespaced, user facing Custom Resource Definition representing a request for a bucket.  OBCs reference a StorageClass to differentiate between provisioners, in the same manner as a PVC would.  Depending on the StorageClass, they bucket may be created dynamically or preexisting.  In the later case, new credentials are created and attached to the existing bucket.
```

Comment:
_- ObjectBucketClaim: a namespaced, user facing Custom Resource Definition representing a request for a bucket.  OBCs reference a StorageClass to differentiate between provisioners, in the same manner as a PVC would.  Depending on the StorageClass, **they(( bucket may be created dynamically or preexisting. _
- ObjectBucketClaim (OBC): a namespaced, user facing Custom Resource Definition representing a request for a bucket.  OBCs reference a StorageClass to differentiate between provisioners, in the same manner as a Persistent Volume Claim (PVC) would.  Depending on the StorageClass, the bucket may be created dynamically or preexisting.